src/app/dashboard/dashboard-resolve.service.test.ts

```diff
@@ -1,35 +1,40 @@
-import {async, TestBed} from '@angular/core/testing';
-
+import { Observable } from 'rxjs/Rx';
+import { fakeAsync, tick } from '@angular/core/testing';
 import { DashboardResolveService } from './dashboard-resolve.service';
-import { DashboardService } from './dashboard.service';
-
-const observableMock = {};
-
-class DashboardServiceMock {
-    getWidgets() { return observableMock; }
-}
 
 describe('dashboard-resolveService', () => {
     let sut;
-    let dashboardService;
-
-    beforeEach(async(() => {
-        TestBed.configureTestingModule({
-            providers: [
-                {provide: DashboardService, useClass: DashboardServiceMock },
-                DashboardResolveService
-            ]
-        })
-        .compileComponents()
-        .then(() => {
-            sut = TestBed.get(DashboardResolveService);
-            dashboardService = TestBed.get(DashboardService);
-            spyOn(dashboardService, 'getWidgets').and.callThrough();
-        });
-    }));
+    let DashboardService;
+    let ToastsManager;
+
+    beforeEach(() => {
+
+        DashboardService = {
+            getWidgets: jasmine.createSpy('getWidgets')
+        };
+
+        ToastsManager = {
+            error: jasmine.createSpy('error')
+        };
+
+        sut = new DashboardResolveService(DashboardService, ToastsManager);
+    });
 
     it('should call dashboardService getWidgets method', () => {
+        DashboardService.getWidgets.and.returnValue(Observable.create());
         sut.resolve();
+
         expect(sut.dashboardService.getWidgets).toHaveBeenCalled();
     });
+
+
+    xit('should cut widgets stream on error', fakeAsync(() => {
+        const error = 'Fake error';
+
+        DashboardService.getWidgets.and.returnValue(Observable.throw(error));
```

Comment:
Help me please with testing errors on Observables. How to mock fail response from DashboardService.getWidgets for example?